Mem cache

jgallagher59701 opened the pull request:
The ObjMemCache - now in bes/dap - implements a [memory cache](https://docs.google.com/a/opendap.org/document/d/1ewY9y2Uiiv4pBkl2msn-3BLUZW-QxfkCyOfWaxlLI7U/edit?usp=sharing). I've used this with the netcdf handler and while that's not the best test case, it does produce faster object access times (about 10+ times faster) with little measurable overhead. Unfortunately, the netcdf handler is a submodule, so you'll need to look at that code separately. See the mem_cache branch there. I may add this to the hdf4/5 handlers if there's time.

kyang2014 commented:
So the memory cache is still not part of BES?


From: James Gallagher [mailto:notifications@github.com]
Sent: Tuesday, May 24, 2016 3:13 PM
To: OPENDAP/bes
Subject: [OPENDAP/bes] Mem cache (#24)


The ObjMemCache - now in bes/dap - implements a memory cache<https://docs.google.com/a/opendap.org/document/d/1ewY9y2Uiiv4pBkl2msn-3BLUZW-QxfkCyOfWaxlLI7U/edit?usp=sharing>. I've used this with the netcdf handler and while that's not the best test case, it does produce faster object access times (about 10+ times faster) with little measurable overhead. Unfortunately, the netcdf handler is a submodule, so you'll need to look at that code separately. See the mem_cache branch there. I may add this to the hdf4/5 handlers if there's time.

________________________________
You can view, comment on, or merge this pull request online at:

https://github.com/OPENDAP/bes/pull/24

Commit Summary

  *   Added DDSMemCache
  *   Added DDSMemCache to Makefile.am
  *   checkpoint
  *   The DDSMemCache code is working and tested
  *   Added comments to the DDSMemCache
  *   Changed DDSMemCache to ObjMemCache - a memory cache for DapObj
  *   Simplified names for the ObjMemCache.
  *   gitignore update
  *   The ObjMemCache now frees all of the pointers it holds...
  *   doc update

File Changes

  *   M configure_modules.ac<https://github.com/OPENDAP/bes/pull/24/files#diff-0> (9)
  *   M dap/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-1> (7)
  *   A dap/ObjMemCache.cc<https://github.com/OPENDAP/bes/pull/24/files#diff-2> (158)
  *   A dap/ObjMemCache.h<https://github.com/OPENDAP/bes/pull/24/files#diff-3> (161)
  *   M dap/unit-tests/.gitignore<https://github.com/OPENDAP/bes/pull/24/files#diff-4> (2)
  *   M dap/unit-tests/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-5> (10)
  *   A dap/unit-tests/ObjMemCacheTest.cc<https://github.com/OPENDAP/bes/pull/24/files#diff-6> (234)
  *   M functions/tests/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-7> (4)
  *   M modules/netcdf_handler<https://github.com/OPENDAP/bes/pull/24/files#diff-8> (2)

Patch Links:

  *   https://github.com/OPENDAP/bes/pull/24.patch
  *   https://github.com/OPENDAP/bes/pull/24.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/OPENDAP/bes/pull/24>

jgallagher59701 commented:
On May 24, 2016 at 15:23:46, kyang2014 (notifications@github.com) wrote:

So the memory cache is still not part of BES?
It is part of the BES - it’s in bes/dap. Use git to checkout the branch ‘mem_cache’ to see the code or wait for me to merge it. Each handler has to be modified a little to use the cache.

James




From: James Gallagher [mailto:notifications@github.com]
Sent: Tuesday, May 24, 2016 3:13 PM
To: OPENDAP/bes
Subject: [OPENDAP/bes] Mem cache (#24)


The ObjMemCache - now in bes/dap - implements a memory cache<https://docs.google.com/a/opendap.org/document/d/1ewY9y2Uiiv4pBkl2msn-3BLUZW-QxfkCyOfWaxlLI7U/edit?usp=sharing>. I've used this with the netcdf handler and while that's not the best test case, it does produce faster object access times (about 10+ times faster) with little measurable overhead. Unfortunately, the netcdf handler is a submodule, so you'll need to look at that code separately. See the mem_cache branch there. I may add this to the hdf4/5 handlers if there's time.

________________________________
You can view, comment on, or merge this pull request online at:

https://github.com/OPENDAP/bes/pull/24

Commit Summary

* Added DDSMemCache
* Added DDSMemCache to Makefile.am
* checkpoint
* The DDSMemCache code is working and tested
* Added comments to the DDSMemCache
* Changed DDSMemCache to ObjMemCache - a memory cache for DapObj
* Simplified names for the ObjMemCache.
* gitignore update
* The ObjMemCache now frees all of the pointers it holds...
* doc update

File Changes

* M configure_modules.ac<https://github.com/OPENDAP/bes/pull/24/files#diff-0> (9)
* M dap/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-1> (7)
* A dap/ObjMemCache.cc<https://github.com/OPENDAP/bes/pull/24/files#diff-2> (158)
* A dap/ObjMemCache.h<https://github.com/OPENDAP/bes/pull/24/files#diff-3> (161)
* M dap/unit-tests/.gitignore<https://github.com/OPENDAP/bes/pull/24/files#diff-4> (2)
* M dap/unit-tests/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-5> (10)
* A dap/unit-tests/ObjMemCacheTest.cc<https://github.com/OPENDAP/bes/pull/24/files#diff-6> (234)
* M functions/tests/Makefile.am<https://github.com/OPENDAP/bes/pull/24/files#diff-7> (4)
* M modules/netcdf_handler<https://github.com/OPENDAP/bes/pull/24/files#diff-8> (2)

Patch Links:

* https://github.com/OPENDAP/bes/pull/24.patch
* https://github.com/OPENDAP/bes/pull/24.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/OPENDAP/bes/pull/24>
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub


-- 
James Gallagher
jgallagher@opendap.org
